### ISSUE TITLE
navigating back from multiple intercepted route breaks routing

### ISSUE THREAD

**Issue by roksui** [vercel/next.js#66853]
### Link to the code that reproduces this issue

https://github.com/roksui/my-nextagram

### To Reproduce

1. Clone the repository and checkout to the master branch.
2. Start the application in dev mode using `npm run dev`.
3. Click on `View my notifications` link.
4. Click on `View photo 1`.
5. Do a browser refresh (pressing F5).
6. Press the browser back button (once or twice).
7. The url changes to the previous route but the page does not.

### Current vs. Expected behavior

When I press the back button in step 6 above, I expected the browser to navigate back to the intercepted route for `/notifications`. However, only the URL changes and the page stays the same, showing the intercepted route of `/photos`.

### Provide environment information

```bash
Operating System:
  Platform: win32
  Arch: x64
  Version: Windows 10 Pro
  Available memory (MB): 32545
  Available CPU cores: 16
Binaries:
  Node: 18.17.1
  npm: N/A
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.0-canary.29 // Latest available version is detected (15.0.0-canary.29).
  eslint-config-next: 15.0.0-canary.29
  react: 19.0.0-rc-f994737d14-20240522
  react-dom: 19.0.0-rc-f994737d14-20240522
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Parallel & Intercepting Routes

### Which stage(s) are affected? (Select all that apply)

next dev (local), next start (local)

### Additional context

Basically my use case is to open two modals A and B, where modal B is opened from modal A. I have it structured as:
```
app
  @modal
    (.)notifications
    (.)photos
  notifications
    page.tsx
  photos
    page.tsx
```
I could not find any documentation on implementing multiple intercepted routes, so not sure if this structure is the way to go.

With this structure, the intercepting works with no problems with the two modals. Also, refreshing on the first intercepting route and navigating back works too. However, refreshing on the second intercepting route and navigating back causes an issue.

**Comment by hypervisor** [comment]
Can confirm this happens to my app as well, in both dev and prod.

**Comment by khuezy** [comment]
Related, when using `<Link />` to link to base of the modal, the modal still persists when it should be destroyed instead.
eg `/photos/A`, then use `Link` to go to `/photos`, the route changes but the modal is still there.

**Comment by khuezy** [comment]
Try duplicating `default.tsx` and rename duplicated file to `page.tsx` for each `@modal` folder. That should clear the modal when you go back to the root of each intercepted folder.

**Comment by vanlooverenkoen** [comment]
I am having the same issue on `14.2.4` 

https://github.com/vercel/next.js/assets/21172855/7c32d716-5877-40ed-9963-7473fa897b1d

**Comment by tenghuan123** [comment]
> I am having the same issue on 我在`14.2.4`
> 
> My flow goes like this我的流程是这样的
> 
> * page A 页面 A
> * modal 资本
> * page B B页
> 
> (When going back it should look like this)（回去时，它应该看起来像这样）
> 
> * page B B页
> * modal 资本
> * page A 页面 A
> 
> This is what I get:这是我得到的：
> 
> * page B B页
> * modal 资本
> * page a (but modal is still shown and not removed from the UI)页面 a（但模式仍显示，并且未从 UI 中删除）
> 
>  Screen.Recording.2024-07-02.at.14.15.14.mov

You need to add [... catchAll] to capture what UI should be displayed when the user leaves the interceptor route. Just like this:
![image](https://github.com/user-attachments/assets/c108f0c1-70bd-423d-9cc2-97573594ffa3)
![image](https://github.com/user-attachments/assets/6939aca6-4337-4d72-933b-67203a8ac201)

**Comment by ljuborados** [comment]
It's broken with a catchAll route as well.

Also as a bonus, it breaks 404 handling for soft-navigation (just add a Link pointing to `/test`) and it'll happily navigate to it while showing the old page - no errors, no anything.

**Comment by dragos99** [comment]
> Try duplicating `default.tsx` and rename duplicated file to `page.tsx` for each `@modal` folder. That should clear the modal when you go back to the root of each intercepted folder.

Doesn't work. Same issue :(

**Comment by marcoshernanz** [comment]
Thanks for the reproduction. I can reproduce the stale-page behavior on `15.0.0-canary.29` and `15.5.25`, but not on `16.3.4` or `16.4.0-canary.13`. After refreshing `/photos/1`, Back now renders content for `/notifications` instead of leaving the photo page under the previous URL.

This appears to have been resolved by the router restoration changes in #87135, first released in `16.1.0-canary.27`. Closing as fixed. If the original stale-page behavior still occurs on a current release, please open a new issue with an updated reproduction.
